### DISCUSSION TITLE
POS integrations

### DISCUSSION THREAD

**Discussion by brandonkelly** [craftcms/commerce#2287] [Ideas]
Could be done with a general purpose card reader (e.g. https://www.magtek.com/product/pin-pads), and/or we can integrate with Square (https://squareup.com/developers, https://docs.connect.squareup.com/api/connect/v1/#navsection-transactions) - probably as a separate plugin, since the workflow/settings involved in integrating Square webhooks will be more complex than normal gateways.

**Comment by andris-sevcenko** [comment]
Not sure how we would integrate with this. Square has their own POS solution, but if this would be to allow Commerce to act as a POS, then this would come as a REST API with individual plugins bridging the gap between any POS and Commerce

**Comment by brandonkelly** [comment]
[Stripe Terminal](https://stripe.com/terminal) may also be an option, which is now out of beta.

**Comment by j2is** [comment]
Commonly asked for synchronising inventory between online purchases and in-store, and being able to look up online or physical orders in Craft CMS. Smaller requests: putting gift cards or refunds through on a POS  and having that reflected in Craft. Square might be complex for sure, I'd vote for Stripe terminal

**Comment by lindseybradford** [comment]
I don't want to assume that because this is an open issue, that it's on the roadmap: is POS inventory management a feature that's underway, and if so, roughly when could we expect to see it drop?

**Comment by lukeholder** [comment]
@lindseybradford No timeline sorry, but not in the immediate future.

**Comment by j2is** [comment]
This one would be a game changer. Patiently await the day this becomes a reality

**Comment by aeu** [comment]
Hi, it's been approximately a year since the last comment on this thread.  Is anything in the works as far as integrating Craft Commerce & POS ?

**Comment by nfourtythree** [comment]
Hi All

There are a lot of variations when it comes to POS, especially when it comes to what people would like to see Commerce handle and how gateways/POS service providers handle things.

It would be great if people could outline what they are looking for in terms of this feature.

- POS feature you would like to see in Commerce
- Third party POS services/gateways to integrate with
- Any other POS information

This will give us more information on what a POS feature in Commerce could look like.

Thanks!

**Comment by stevenjwright** [comment]
@nfourtythree I have a client who's seeking this feature in the next 12 months. 

**Use Case**
- Single Brick & Mortar store with international E-Comm presence.  
- Current POS lacks API functions to keep stock in sync with site
- Primary reason for this is the ability for in-sync stock between Site & Phycical Store.
- E-Comm uses Stripe, Physical Store uses WorldPay Terminal & Cash

**Would be nice Features**
- Multi-(Physical) Store Stock / Inventory
- Sales Reporting (Breakdown per store/location/online v physical)
- Some Products to be POS-Only not online, and other way around (though could be a custom field)
- Some ability for a USB-based Barcode Scanner
- Email/SMS Receipts

Hope that helps!

**Comment by andykearney** [comment]
@nfourtythree we might have some useful insights here. 

We're a single bricks and mortar retail store with a decent amount of online sales, a workshop element and a b2b distribution business. All managed exclusively through craft commerce. We built an inventory management plugin to handle inventory tracking and purchasing and the workshop side. It's all about as complex as these things get. 

We've struggled with integrating inventory systems for a long while so abandoned it in favour of an all in one solution.

We essentially use the CP as a POS which does work, albeit with a few opportunities for improving efficiency and a few missing features. Once commerce is the source of truth as far as inventory is concerned then there's no real need to want to integrate with an external POS. This is only really possible if you can track inventory properly. If you already have a POS that is the source of truth for your inventory, then sending stock info to commerce should be relatively easy if there's a decent API. Most POS platforms priority is to keep you in their system so decent API's are rare in our experience.

A few things spring to mind that would make the CP function much better as a POS

- Customer/User relationship is super confusing for staff in practice.
- As soon as you have a customer that doesn't want to give you an email address, things get complicated. 
- Printing/scanning barcode labels for variants. 
- Full or partial payments via multiple payment methods is a daily encounter and works fine now. Full or partial refunds to a payment method that's different to the source is essential. This should include credit notes/customer credit accounts. 
- Integrate with card readers for in person payments. Stripe do offer this now so we'll likely switch and figure it out. We currently us a manual gateway and an external card reader provider. The authorise/capture workflow makes it clunky.
- Better way of handling deposits. Really needs customer accounts rather than just being a partial payment on a specific order.
- Integrated commerce specific reporting. We often found reports can be useful in the moment when dealing face to face with customers. 

Theres probably loads more. But most of it really relies on proper inventory tracking and PO management first.

Otherwise, as far as I can see everything needed to integrate with another POS platform and design a commerce site around it  already exists. The weak point will always be the capability of the API.

**Comment by aeu** [comment]
Hi @nfourtythree 

What we basically need is to be able to have a computer (or tablet), a credit card machine, a printer and a barcode scanner at a counter.  Cashier scans items, payment gets processed, inventory updated and receipt printed.   Just like a cash register.
